drivers/usb/usbstor/scsi.c

```diff
@@ -444,9 +468,20 @@ USBSTOR_CBWCompletionRoutine(
 
         if (!Mdl)
         {
-            DPRINT1("USBSTOR_DataTransfer: Mdl - %p\n", Mdl);
+            DPRINT1("USBSTOR_CBWCompletionRoutine: Mdl - %p\n", Mdl);
             goto ResetRecovery;
         }
+    }
+    else
+    {
+        TransferBuffer = Request->DataBuffer;
+
+        if (!Request->DataBuffer)
+        {
+            DPRINT("USBSTOR_CBWCompletionRoutine: Request->DataBuffer == NULL!\n");
+            return STATUS_INVALID_PARAMETER;
```

Comment:
This is not a good return code for a completion routine. It should be `STATUS_CONTINUE_COMPLETION`... and the real status may need to go in `Irp->IoStatus.Status`